hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/AbstractSqmAttributeJoin.java

```diff
@@ -141,4 +141,8 @@ public JoinType getJoinType() {
 		return getSqmJoinType().getCorrespondingJpaJoinType();
 	}
 
+	@Override
+	public Integer getTupleLength() {
+		return getLhs().get( getAttribute().getName() ).getTupleLength();
```

Comment:
Is that the only place where we could have an embeddable "nested" within an expression? IMO, instead of creating an `SqmPath` object as a side-effect here, it would be better to extract the code from `SqmEmbeddedValuedSimplePath#getTupleLength` to a helper method and base it on the domain model only i.e. no calls to `get`.

Reply:
Well, wouldn't the `SqmPath`s be resolved anyways down the road, taking advantage of the `reusablePaths` map? I thought that was the case, as this is already used for `SqmEmbeddedValuedSimplePath#getTupleLength`.

Either way, any suggestions on how to _base it on the domain model only_? At the time I thought about it but it wasn't clear to me if we had something like a "tuple length" information in the domain model already.

Reply:
@beikov switched to a `DomainModel#getTupleLength()`  solution, like suggested.